### PR TITLE
Docs refactor

### LINKED ISSUE #3559: [Discussion] Issues with new code commenting style
Apologies for posting this here rather than in the [comment style change PR](https://github.com/tux3/qTox/pull/3536), I didn't see it until a few hours ago when it was already merged.

I'm in the processing of rebasing my code to the style introduced in PR #3536 when I found a few issues with the new style. First of all I would like to state that evaluating pros/cons, I agree with putting comments in the source files for **most cases**, but not all.

In particular, I would like to make the argument against putting comments for class descriptions (e.g. comments tagged with `@class`) and comments for variable descriptions (e.g. comments tagged with `@var`) in source files.

First of all, class comments basically declare the outline of a class, the functionality and the intended purpose or role for the class. These concepts usually do not change very often making header-triggered recompilation times minimal compared to other changes. Furthermore, it is very likely that a documentation change for the class changes at least some part of the header as well (whether this is a new function, refactoring of existing functions or the change in the parameters of certain functions) which would result in header-trigged recompliation regardless of where the documentation is placed.

By the change in PR #3536 class documentations just kinda float around the top of source files, making getting to them harder (functions are fine since QtCreator has a "Follow Symbol Under Cursor" option that takes you between the source and header for the function, allowing easy preview). Additionally, in the case of nested classes or multiple classes in one header, things get weird (do you put both class definitions at the top of the file? Or do you put the second/nested class documentation at the start of it's function implementions/constructors).

I've mentioned nested classes specifically because in one of my PRs I have a case of a nested class which is completely delared in the header (no functions declared in source). Do I isolate the class documention from the rest of the class and put it in the source file? That seems kinda strange.

Moving on to variable documentation, I believe that member variable documention must be kept right at the point of declaration _because_ of it's volatility. The fact that members are separated from their documentation, in my opinion, will only lead to changes in name/usage/type/role of the member variable that becomes inconsistent with the documentation due to failure to update documentation (people might not notice there is documentation for the particular variable for example). The same idea applies to nested/multiple classes, where do those variable documentations go?

Again, I made this point when I was rebasing my PR to the changes in master, having to cross-reference repeatedly between the header and source whether a particular variable was changed, renamed or deleted (within my PR).

A third but minor point regards the style of commenting used. From the codebase I've seen:

```
/**
 * Comment
 */
```

and

```
/// Comment
```

styles being used, in which the former is what QtCreator defaults to (at least on my system). The style of comment implemented by the PR is:

```
/**
Comment
*/
```

Which can easily be converted into the first style. Using the first style at least enforces consistency as developers won't have to worry about deliberately changing the comment style, having QtCreator automatically generate a valid template that's consistent with rest of the comments in the codebase.

Besides, even after the PR, I still see many classes who's comments have not been affected by the PR which use the first style mentioned above. Wouldn't just sticking to the first style allow for better integration? At least commenting styles would be consistent across the codebase whilst comment content can slowly be improved/added.

I've only brought these issues up now because comment consistency is something I deem extremely important, and I want to change my existing PR code to satisfy the "qTox" comment style (whether I agree with it or not). But before doing so, want to share a bit of my rationale regarding appropriate comment style.

---

**tl;dr** I believe `@class` and `@var` comments should be in headers and we should use the:

```
/**
 * Comment
*/
```

style of comments.

@Diadlo @sudden6 @zetok

### PR DESCRIPTION
**Pull request by Diadlo**
Fix #3559.

---

This change is [Reviewable](https://reviewable.io/reviews/tux3/qtox/3577)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -214,40 +214,40 @@ Don't put docs in .h files, if there is a corresponding .cpp file.
 #include "blabla.h"
 
 /**
-I can be briefly described as well!
-*/
+ * I can be briefly described as well!
```
Comment (initramfs) [inline CONTRIBUTING.md]: Function statements can have `@brief` tags infront of the first line of descriptions too. Also perhaps demonstrate a function with parameters and a return statement, e.g:

``` c++
/**
 * @brief I can be briefly described as well!
 *
 * And here goes my longer descrption!
 *
 * @param x Description for the first parameter
 * @param y Description for the second paramater
 * @return An amazing result
 */
static int example(int x, int y)
{
    // Function implementation...
}
```

**Comment by zetok** [comment]
Since osx travis was broken for a bit, you will need to rebase your PR on newest master to get it to pass travis check. Sorry for inconvenience.

**Comment by antis81** [comment]
Review status: 0 of 66 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

_[src/audio/audio.cpp, line 348 [r2]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KOkKNn74FBHdPtHSG28:-KOkKNn8468tqPvDHlCc:baonwh8) ([raw file](https://github.com/qtox/qtox/blob/879753cf2037b213d8e14c9be196e544e710a576/src/audio/audio.cpp#L348)):_

> ``` cpp
>     assert(!alInDev);
> 
>     /// @todo Try to actually detect if our audio source is stereo
> ```

Can you please cange this to

``` c++
// TODO: Try to actually detect if our audio source is stereo
```

---

_[src/core/core.cpp, line 271 [r2]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KOkKuI_Y2PNTRtibc9b:-KOkKuIazJj71iCsEuJu:b-lgnuc0) ([raw file](https://github.com/qtox/qtox/blob/879753cf2037b213d8e14c9be196e544e710a576/src/core/core.cpp#L271)):_

> ``` cpp
> 
>     /// @todo NOTE: This is a backwards compatibility check,
>     /// once most people have been upgraded away from the old HistoryKeeper, remove this
> ```

Those lines are not part of the documentation, but "internal" code comments. Can you please change those "TODO:" comments to just normal comments?

``` c++
// TODO: This is a backwards compatibility check,
// once most people have been upgraded away from the old HistoryKeeper, remove this
```

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3577)_

**Comment by Diadlo** [comment]
@antis81 Why you prefer `TODO:` instead of `@todo`?

**Comment by antis81** [comment]
> Why you prefer TODO: instead of @todo?

I already wrote in the review and it is not a personal preference, but it is actually wrong. Inlined code comments are not part of any (official) documentation and should not appear as such. They are just regular inline comments to outline a logical or mathematical problem to other developers. Those comments are rare and coders should strive to finish implementation instead. Some editors, like e.g. Qt Creator or Eclipse and even VS have integrated "todo list" support, which allows to list inline comments starting with `TODO:`, `FIXME:`, etc. to help finding (and eliminating) those places. A `// NOTE:` comment is redundant -- a comment already is a note. IOW: Inline comments should be completely avoided unless they carry important information to outline a specific decision to other developers.

**Comment by iphydf** [comment]
For future reference: I'm going to ignore 3000 line diffs in a single PR unless someone convinces me that it really can't be broken into pieces.

**Comment by Diadlo** [comment]
@iphydf You think it should be splited on many same PR with different scope?)
Or what?

**Comment by iphydf** [comment]
I would prefer to see one PR that _just_ adds leading stars. That can be a huge but trivial diff, easy to check. If I need to check every line manually to see whether it was just stars added or also other stuff changed, I won't even begin looking at it.

**Comment by sudden6** [comment]
Reviewed 37 of 66 files at r1, 5 of 11 files at r2, 9 of 18 files at r3, 8 of 11 files at r4.
Review status: 59 of 65 files reviewed at latest revision, 7 unresolved discussions.

---

_[src/persistence/settings.cpp, line 667 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOlHX7YJxZulsDQVvd:-KPOlHX8w4dMuyC70TRb:b-fqkn5m) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/persistence/settings.cpp#L667)):_

> ``` cpp
> #else
>     /**
>      * TODO: Change QStandardPaths::DataLocation to AppDataLocation when upgrate Qt to 5.4+
> ```

should this really be in api doc?

---

_[src/video/videoframe.cpp, line 28 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOjw6DjYrnkudGatts:-KPOjw6DjYrnkudGattt:be26zji) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/video/videoframe.cpp#L28)):_

> ``` cpp
> 
> /**
> <<<<<<< c05d0f91224909f32c6a1da4a0c55ad92dd0637c
> ```

git leftover

---

_[src/widget/form/settings/aboutform.cpp, line 54 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOmBskljFF91UpnkGy:-KPOmBsl4XE8_AIP72-L:b-7tlysh) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/widget/form/settings/aboutform.cpp#L54)):_

> ``` cpp
> 
> /**
>  * TODO: When we finally have stable releases: build-in a way to tell
> ```

this shouldn't be in api doc

---

_[src/widget/form/settings/generalform.cpp, line 270 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOmOhr0T0rVisZbhm7:-KPOmOhr0T0rVisZbhm8:b-fqkn5m) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/widget/form/settings/generalform.cpp#L270)):_

> ``` cpp
>     connect(bodyUI->cbGroupchatPosition, &QCheckBox::stateChanged, this, &GeneralForm::onGroupchatPositionChanged);
> 
>     /// prevent stealing mouse wheel scroll
> ```

should this really be in api doc?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3577)_

**Comment by Diadlo** [comment]
Review status: 55 of 64 files reviewed at latest revision, 7 unresolved discussions.

---

_[src/persistence/settings.cpp, line 667 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOlHX7YJxZulsDQVvd:-KPSPkQl52aea4ykRH2b:b-ascj6f) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/persistence/settings.cpp#L667)):_
<details><summary>

<i>Previously, sudden6 wrote…</i></summary>



> should this really be in api doc?
> </details>
> It will not be in api doc

---

_[src/widget/form/settings/aboutform.cpp, line 54 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOmBskljFF91UpnkGy:-KPSPyYlftOdwqGhMb6n:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/widget/form/settings/aboutform.cpp#L54)):_
<details><summary>

<i>Previously, sudden6 wrote…</i></summary>



> this shouldn't be in api doc
> </details>
> Done.

---

_[src/widget/form/settings/generalform.cpp, line 270 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOmOhr0T0rVisZbhm7:-KPSPztui91BWrS2cE0z:b-ascj6f) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/widget/form/settings/generalform.cpp#L270)):_
<details><summary>

<i>Previously, sudden6 wrote…</i></summary>



> should this really be in api doc?
> </details>
> It will not be in api doc

---

_[src/video/videoframe.cpp, line 28 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOjw6DjYrnkudGatts:-KPSQ1dHbSGAG6l-OsW4:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/video/videoframe.cpp#L28)):_
<details><summary>

<i>Previously, sudden6 wrote…</i></summary>



> git leftover
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3577)_

**Comment by sudden6** [comment]
Reviewed 1 of 66 files at r1, 1 of 18 files at r3, 6 of 6 files at r5.
Review status: 62 of 64 files reviewed at latest revision, 5 unresolved discussions.

---

_[src/persistence/settings.cpp, line 667 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOlHX7YJxZulsDQVvd:-KPTyzWapKWtZ4L73ZdG:bd1rym1) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/persistence/settings.cpp#L667)):_
<details><summary>

<i>Previously, Diadlo (Diadlo) wrote…</i></summary>



> It will not be in api doc
> </details>
> why the doxygen comment then?

---

_[src/widget/form/settings/generalform.cpp, line 270 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOmOhr0T0rVisZbhm7:-KPU-ZTLrBvMK6m5Z6k_:b-12ewjx) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/widget/form/settings/generalform.cpp#L270)):_
<details><summary>

<i>Previously, Diadlo (Diadlo) wrote…</i></summary>



> It will not be in api doc
> </details>
> why the tripple `/` then ?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3577)_

**Comment by sudden6** [comment]
seems like a few non docs refactor commits got in there

---

Reviewed 3 of 11 files at r6, 1 of 2 files at r7.
Review status: 61 of 69 files reviewed at latest revision, 5 unresolved discussions.

---

_[src/persistence/settings.cpp, line 667 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOlHX7YJxZulsDQVvd:-KPUEFEv0xUNp886VjQQ:bwu8ixj) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/persistence/settings.cpp#L667)):_
<details><summary>

<i>Previously, sudden6 wrote…</i></summary>



> why the doxygen comment then?
> </details>
> OK now

---

_[src/widget/form/settings/generalform.cpp, line 270 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3577#-KPOmOhr0T0rVisZbhm7:-KPUFW0QHKwnIVaUHRwN:bwu8ixj) ([raw file](https://github.com/qtox/qtox/blob/01af22e28fa03facbb12d82ed8e61c1a8deacbd9/src/widget/form/settings/generalform.cpp#L270)):_
<details><summary>

<i>Previously, sudden6 wrote…</i></summary>



> why the tripple `/` then ?
> </details>
> OK now

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3577#-:-KPUFcW0HzYW-_sb6hsp:beqefxt)_

**Comment by sudden6** [comment]
:lgtm:

---

Reviewed 1 of 2 files at r7, 5 of 11 files at r8.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3577#-:-KPWW_gEcX11LBQCaahR:bnfp4nl)_

**Pull request merged**